From 53a0b421192173388d3a74b2e33f8d5ad89f7c90 Mon Sep 17 00:00:00 2001
From: Min Lin <linmin@eswincomputing.com>
Date: Tue, 30 Sep 2025 14:08:15 +0800
Subject: [PATCH 524/541] fix(dmac,llc):Remove redudant call of suspend in dmac
 probe

Changelogs:
1.Removed axi_dma_suspend(chip) from dw_probe(). This is redudant
  and will cause uncorrect refcnt for tbu, clk and reset.
2.Move win2030_tbu_power(&pdev->dev, true) to the bottom of the
  llc_probe

Signed-off-by: Min Lin <linmin@eswincomputing.com>
---
 drivers/dma/dw-axi-dmac/dw-axi-dmac-platform.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/drivers/dma/dw-axi-dmac/dw-axi-dmac-platform.c b/drivers/dma/dw-axi-dmac/dw-axi-dmac-platform.c
index 1e1cd6acf233..0dbb26cd0afd 100644
--- a/drivers/dma/dw-axi-dmac/dw-axi-dmac-platform.c
+++ b/drivers/dma/dw-axi-dmac/dw-axi-dmac-platform.c
@@ -1824,9 +1824,6 @@ static int dw_probe(struct platform_device *pdev)
 
 	dev_info(chip->dev, "DesignWare AXI DMA Controller, %d channels\n",
 		 dw->hdata->nr_channels);
-	ret = axi_dma_suspend(chip);
-	if (ret < 0)
-		goto err_pm_disable;
 
 	dw_axi_dma_register_debugfs(chip);
 
-- 
2.54.0

